src/Agent.php

```diff
@@ -61,18 +72,48 @@ public function getCollector(string $name): DataCollector
         return $this->collectors->get($name);
     }
 
+    /**
+     * We need to keep track of the current Transaction so the app can access it for
+     * distributed tracing and other tasks. For now, we expect a single transaction
+     * to be sufficient for HTTP requests and jobs. This will need to change if we
+     * ever start allowing nested transactions.
+     */
+    public function hasCurrentTransaction(): bool
+    {
+        return null !== $this->current_transaction;
+    }
+
+    public function setCurrentTransaction(Transaction $transaction): void
+    {
+        $this->current_transaction = $transaction;
+    }
+
+    public function clearCurrentTransaction(): void
+    {
+        $this->current_transaction = null;
+    }
+
+    public function currentTransaction(): Transaction
+    {
+        if (null === $this->current_transaction) {
```

Comment:
Should we reuse the existing method instead?

```suggestion
        if (!$this->hasCurrentTransaction()) {
```

Reply:
I always feel a little odd negating a method within the class like that. IMHO, it's easier to miss the leading `!` then the logical comparison. So when I'm dealing with class internals, I often just use the raw comparison. But that's my issue and I think makes sense to use the method when stop to consider. This may be a "code smell" that says you need a named method for the other case. Something to think about.

Anyway, I switched to the suggestion.